### PR TITLE
fix(integration): test_metric_longhorn_snapshot_actual_size_bytes

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/7532

### CODE REVIEW

**Comment by c3y1huang** [comment]
Test result: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5825/

Note: Running regression with the current master branch code base will fail due to https://github.com/longhorn/longhorn/issues/6563#issuecomment-1875159217. so testing with at commit 91bc9cb701ac3116d37311fb8c732ac049a560f4.

**Review by yangchiu — DISMISSED** [review]
LGTM

File: manager/integration/tests/test_metric.py [hunk 1]
```diff
@@ -158,6 +161,21 @@ def filter_metric_by_labels(metrics, labels):
         assert total_metrics["value"] >= 0.0
 
 
+def wait_for__metric_count_all_nodes(client, core_api, metric_name, metric_labels, expected_count): # NOQA
```
Comment (innobead) [inline manager/integration/tests/test_metric.py:164]: `wait_for_metric_count_all_nodes`

File: manager/integration/tests/test_metric.py [hunk 2]
```diff
@@ -158,6 +161,21 @@ def filter_metric_by_labels(metrics, labels):
...
+        time.sleep(RETRY_INTERVAL)
+
+        try:
+            check_metric_count_all_nodes(client, core_api, metric_name,
+                                         metric_labels, expected_count)
+            break
+        except AssertionError:
+            continue
+
+    check_metric_count_all_nodes(client, core_api, metric_name,
```
Comment (innobead) [inline manager/integration/tests/test_metric.py:175]: Why need to run this check again?

**Review by innobead — CHANGES_REQUESTED** [review]

File: manager/integration/tests/test_metric.py (hunk 2 above)
Comment (c3y1huang) [reply manager/integration/tests/test_metric.py:175]: We could also `raise AssertionError("Cannot find the metric matching the labels")`, but having another run would reveal the specific assertion error, I think this way gives a bit more clarity.
```
>       assert len(filtered_metrics) == expected_count
E       assert 1 == 4
E         -1
E         +4
```

**Review by innobead — APPROVED** [review]

**Pull request merged**